### ISSUE TITLE
Draft ProductList showing up in list of Children

### ISSUE THREAD

**Issue by jeffsnyder** [iFixit/react-commerce#1364]
For some reason, [Nut Drivers](https://www.ifixit.com/Tools/Nut_Drivers) ([Strapi](https://strapi.ifixit.com/admin/content-manager/collectionType/api::product-list.product-list/2388)) is showing up in the list of Children on [Drivers and Wrenches](https://www.ifixit.com/Tools/Drivers_and_Wrenches) ([Strapi](https://strapi.ifixit.com/admin/content-manager/collectionType/api::product-list.product-list/2285)) despite being in Draft _and_ being marked `ExcludeFromHierarchyDisplay = True`. 

![image](https://user-images.githubusercontent.com/7053450/218615349-b9b3f1a2-737b-4a7f-8023-4f3338e19b9f.png)

I made `ExcludeFromHierarchyDisplay = False` and I saw no change. 

I checked another similar situation ([Apple Thunderbolt Display](https://strapi.ifixit.com/admin/content-manager/collectionType/api::product-list.product-list/589?plugins[i18n][locale]=en) is in Draft but is a Child of [Mac Parts](https://www.ifixit.com/Parts/Mac)) but could not find the same behavior/bug. 

I tried publishing and unpublishing the ProductList. No change. 

I tried removing the Parent/Child relationship. That worked! But when the Draft page was added back to the Parent/Child relationship it showed up again in the list of Children. 

I'm going to remove the Parent/Child relationship until you can test/fix. 

![image](https://user-images.githubusercontent.com/7053450/218616092-879af0e0-8dd2-4507-a34c-face3d9322f1.png)

CC @ifixitmaxb @dhmacs

**Comment by dhmacs** [comment]
I found out the cause of this, it's a [bug](https://github.com/strapi/strapi/issues/15682) of the newly deployed Strapi version.
The issue says that the bug will be fixed with `v4.6.2` which is due this week. There are workarounds, but since the bug fix will be released very soon I'd say let's wait for it and update Strapi once it's out.

Also worth mentioning that `excludeFromHierarchyDisplay` seems not to be used at all in our codebase. I don't remember what scenario we planned to address with this, but it has been there since the beginning of the codebase, maybe we just didn't delete it because we weren't sure Strapi migrations would handle the removal correctly

**Comment by jeffsnyder** [comment]
> Also worth mentioning that excludeFromHierarchyDisplay seems not to be used at all in our codebase. 

@dhmacs Can we fix it as part of this issue? Or are you saying we don't think this feature will work if/after we build it. 

FYI @ifixitmaxb

**Comment by ifixitmaxb** [comment]
There are certainly use cases where we want to use excludeFromHierarchyDisplay so it'd be nice if we can use it and it "works". As an example use case, we created this collection and all its children, https://www.ifixit.com/Tools/Microsoft_Game_Console, and we intentionally didn't want to link it from /Tools but we did want to connect it to /Tools via breadcrumbs. We are driving traffic to those Xbox Tools pages via SEO but we don't want them, at least for now, to be found via navigation from /Tools.

**Comment by dhmacs** [comment]
> Can we fix it as part of this issue? Or are you saying we don't think this feature will work if/after we build it.

During our previous conversation, @sterlinghirsh and I discovered that the `excludeFromHierarchyDisplay` attribute has existed in Strapi since the beginning but has never been utilized. Therefore, we concluded that it was unnecessary.

However, thanks to @ifixitmaxb pointing out a potential use case, I believe we should reconsider our conclusion. I think that the name `excludeFromHierarchyDisplay` could be misleading. As explained by Max, the purpose is to hide it from parent children but keep it in the breadcrumbs. Thus, I suggest renaming it to `hideFromParent` and implementing the filter in the frontend code.

Thoughts?
